client/network_connect.go

```diff
@@ -6,23 +6,34 @@ import (
 	"github.com/moby/moby/api/types/network"
 )
 
+// NetworkConnectOptions represents the data to be used to connect a container to the
+// network.
+type NetworkConnectOptions struct {
+	EndpointConfig *network.EndpointSettings
+}
+
+// NetworkConnectResult represents the result of a NetworkConnect operation.
+type NetworkConnectResult struct {
+	// Currently empty; placeholder for future fields.
+}
+
 // NetworkConnect connects a container to an existent network in the docker host.
-func (cli *Client) NetworkConnect(ctx context.Context, networkID, containerID string, config *network.EndpointSettings) error {
+func (cli *Client) NetworkConnect(ctx context.Context, networkID, containerID string, options NetworkConnectOptions) (NetworkConnectResult, error) {
```

Comment:
This is a good start; I'm still contemplating if;

- `containerID` should be rolled up into the options
- more controversial; if `NetworkConnect ` should either be moved, or have an equivalent for Container ("container -> connect to network")

Both ways make sense; then again; `NetworkConnect` (for containers) could become `Container Update -> add network / remove network`).